servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/PrematureClosureBeforeResponsePayloadBodyTest.java

```diff
@@ -137,7 +140,12 @@ public void notAllHeadersReceived() throws Exception {
                 "Transfer-Encoding: chunked\r\n" +
                 "Connection: close\r\n");   // no final CRLF after headers
 
-        assertThrows(PrematureChannelClosureException.class, () -> connection.request(connection.get("/")));
+        Throwable t = assertThrows(Throwable.class, () -> connection.request(connection.get("/")));
```

Comment:
consider using `VerificationTestUtils.assertThrows(PrematureChannelClosureException.class, CloseEventObservedException.class, () -> ...)`. it doesn't allow for verification of the intermediate wrapper but maybe it is good enough or can be enhanced?